docs/architecture.md

```diff
@@ -0,0 +1,223 @@
+
+See [test_application/universes/simple.cpp](https://github.com/TheOpenSpaceProgram/osp-magnum/blob/master/src/test_application/universes/simple.cpp) for how a simple universe can be setup.
+
+### Components
+
+Since Universe uses ECS, Satellites are compositions of Components. Universe Components are simple structs prefixed with **UComp**.
+
+Not much is written yet. **UCompVehicle** contains data for vehicles, **UCompPlanet** contains data for planets, and these can be assigned to Satellites.
+
+### Coordinates
+
+Dealing with space means ridiculously long distances and big numbers. Storing positions in floating point is not ideal, since they lose precision the further the value is from zero. Integers don't suffer from this issue and also run consistently across different processors.
+
+Therefore, signed 64-bit integers was chosen as space coordinates. They are also paired with a power-of-2 scale factor to correlate them to meters, where 2^x unit = 1 meter. A scale factor of 10 means that 1024 units = 1 meter.
+
+*Note: multiple scale factors are not yet implemented, all coordinates are fixed to 1024 units = 1 meter.*
+
+### Coordinate Spaces
+
+A bunch of satellites in an N-body simulation have nothing to do with a bunch of satellites landed on a planet. The planet also rotates, which moves everything landed on it too. This calls upon the need for CoordinateSpaces.
+
+CooridnateSpaces groups satellites together to be part of the same reference frame, centered around a parent Satellite. A good example is a planet owning a CooridnateSpace for its landed Satellites, which means CoordinateSpaces should be optionally rotatable (not yet implemented). A patched conics implementation may want a CoordinateSpace around every orbit-able Satellite.
+
+Satellites can be handed off between CoordinateSpaces. This won't be discusseed in detail here; but in short, the process relies on a few queues.
+
+On a technical level, CoordinateSpaces store contiguous buffers of Satellites integer IDs, positions, and other additional data like velocity that may be needed to calculate trajectories. CoordinateSpaces are free to store data any way they want. This allows them to be specialized for whichever operations are performed on them (such as N-body). Their buffers are exposed through **read-only** [strided array views](https://doc.magnum.graphics/corrade/classCorrade_1_1Containers_1_1StridedArrayView.html) to be accessed externally. These are referred to as **CComp** for coordinate components. All CoordinateSpaces must output CCompX, CCompY, and CCompZ as a common interface, which are Cartesian 64-bit signed integers. 
+
+X, Y, and Z are separated, because SIMD computations favor each to be on their own buffer.
```

Comment:
X, Y, and Z coordinate are stored in separate arrays, because SIMD computations favor a Struct of Arrays data layout